{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"catchAndReturn"},{"kind":"text","text":"("},{"kind":"externalParam","text":"_"},{"kind":"text","text":" "},{"kind":"internalParam","text":"element"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","identifier":"doc:\/\/RxSwift\/documentation\/RxSwift\/Observable","preciseIdentifier":"s:7RxSwift10ObservableC","text":"Observable"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":">"}],"languages":["swift"],"platforms":["iOS"]}]},{"kind":"content","content":[{"anchor":"return-value","level":2,"type":"heading","text":"Return Value"},{"type":"paragraph","inlineContent":[{"type":"text","text":"An observable sequence containing the source sequenceâ€™s elements, followed by the "},{"type":"codeVoice","code":"element"},{"type":"text","text":" in case an error occurred."}]}]},{"kind":"parameters","parameters":[{"name":"element","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"Last element in an observable sequence in case error occurs."}]}]}]}],"schemaVersion":{"major":0,"minor":1,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/rxswift\/groupedobservable\/catchandreturn(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable\/catchAndReturn(_:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Continues an observable sequence that is terminated by an error with a single element."}],"kind":"symbol","metadata":{"role":"symbol","title":"catchAndReturn(_:)","roleHeading":"Instance Method","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"catchAndReturn"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"Observable","preciseIdentifier":"s:7RxSwift10ObservableC"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":">"}],"symbolKind":"method","externalID":"s:7RxSwift14ObservableTypePAAE14catchAndReturnyAA0C0Cy7ElementQzGAHF::SYNTHESIZED::s:7RxSwift17GroupedObservableV","extendedModule":"RxSwift","modules":[{"name":"RxSwift"}]},"hierarchy":{"paths":[["doc:\/\/RxSwift\/documentation\/RxSwift","doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable","doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable\/ObservableType-Implementations"]]},"references":{"doc://RxSwift/documentation/RxSwift":{"role":"collection","title":"RxSwift","abstract":[],"identifier":"doc:\/\/RxSwift\/documentation\/RxSwift","kind":"symbol","type":"topic","url":"\/documentation\/rxswift"},"doc://RxSwift/documentation/RxSwift/GroupedObservable/catchAndReturn(_:)":{"role":"symbol","title":"catchAndReturn(_:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"catchAndReturn"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"Observable","preciseIdentifier":"s:7RxSwift10ObservableC"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Self"},{"kind":"text","text":"."},{"kind":"typeIdentifier","text":"Element"},{"kind":"text","text":">"}],"abstract":[{"type":"text","text":"Continues an observable sequence that is terminated by an error with a single element."}],"identifier":"doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable\/catchAndReturn(_:)","kind":"symbol","type":"topic","url":"\/documentation\/rxswift\/groupedobservable\/catchandreturn(_:)"},"doc://RxSwift/documentation/RxSwift/GroupedObservable":{"role":"symbol","title":"GroupedObservable","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"GroupedObservable"}],"abstract":[{"type":"text","text":"Represents an observable sequence of elements that have a common key."}],"identifier":"doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"GroupedObservable"}],"url":"\/documentation\/rxswift\/groupedobservable"},"doc://RxSwift/documentation/RxSwift/Observable":{"role":"symbol","title":"Observable","fragments":[{"kind":"keyword","text":"class"},{"kind":"text","text":" "},{"kind":"identifier","text":"Observable"}],"abstract":[{"type":"text","text":"A type-erased "},{"type":"codeVoice","code":"ObservableType"},{"type":"text","text":"."}],"identifier":"doc:\/\/RxSwift\/documentation\/RxSwift\/Observable","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"Observable"}],"url":"\/documentation\/rxswift\/observable"},"doc://RxSwift/documentation/RxSwift/GroupedObservable/ObservableType-Implementations":{"role":"collectionGroup","title":"ObservableType Implementations","abstract":[],"identifier":"doc:\/\/RxSwift\/documentation\/RxSwift\/GroupedObservable\/ObservableType-Implementations","kind":"article","type":"topic","url":"\/documentation\/rxswift\/groupedobservable\/observabletype-implementations"}}}